spark/src/main/java/org/apache/iceberg/spark/actions/BaseRemoveReachableFilesSparkAction.java

```diff
@@ -80,6 +84,10 @@ public void accept(String file) {
   public BaseRemoveReachableFilesSparkAction(SparkSession spark, String metadataLocation) {
     super(spark);
     this.metadataLocation = metadataLocation;
+    TableMetadata metadata = TableMetadataParser.read(io, metadataLocation);
```

Comment:
Question: since we are reading the version file here, can we already cache `TableMetadata` to avoid reading it twice? Seems like we can replace `metadataLocation` with `metadata` and then use `metadataFileLocation()` if needed.

Reply:
I have cleaned up the code to remove `metadataLocation `